### PR TITLE
V3.0/fix prod issues

### PR DESCRIPTION
**Pull request by leefrank9527**
Fixed issues from production environment of NLNZ.

### CODE REVIEW

File: webcurator-submit-to-rosetta/src/main/java/nz/govt/natlib/ndha/wctdpsdepositor/preprocessor/ArcIndexProcessor.java [hunk 1]
```diff
@@ -45,7 +45,7 @@
 public class ArcIndexProcessor implements PreDepositProcessor {
     private static final Log log = LogFactory.getLog(ArcIndexProcessor.class);
 
-    private static final boolean APPEND_TO_FILE = true;
+    private static final boolean APPEND_TO_FILE = false;
```
Comment (hannakoppelaar) [inline webcurator-submit-to-rosetta/src/main/java/nz/govt/natlib/ndha/wctdpsdepositor/preprocessor/ArcIndexProcessor.java:48]: What issue does this solve? (I'm having trouble seeing the bigger picture)

File: webcurator-webapp/src/main/resources/application-local+mysql.properties [hunk 2]
```diff
@@ -2,11 +2,11 @@
 
 ## Database properties
 databaseType=mysql
-schema.name=DB_WCT
-schema.url=jdbc:mysql://localhost:3306/DB_WCT
+schema.name=DB_WCT_V3
```
Comment (hannakoppelaar) [inline webcurator-webapp/src/main/resources/application-local+mysql.properties:5]: Won't this break existing installations that rely on these defaults?

Comment (leefrank9527) [reply webcurator-webapp/src/main/resources/application-local+mysql.properties:5]: Sorry.  This is my local configuration.  I would update it. Very grateful for reminding me.

File: webcurator-submit-to-rosetta/src/main/java/nz/govt/natlib/ndha/wctdpsdepositor/preprocessor/ArcIndexProcessor.java (hunk 1 above)
Comment (leefrank9527) [reply webcurator-submit-to-rosetta/src/main/java/nz/govt/natlib/ndha/wctdpsdepositor/preprocessor/ArcIndexProcessor.java:48]: This fixed an issue about depositing collections to Rosetta. When depositing to Rosetta, the CDX will be recreated and joined together. Some CDX was missing and failed to deposit to Rosetta.

File: webcurator-webapp/src/main/resources/application-local+mysql.properties (hunk 2 above)
Comment (leefrank9527) [reply webcurator-webapp/src/main/resources/application-local+mysql.properties]: Hi, @hannakoppelaar , I've updated the branch v3.0/fix-prod-issues to change the configuration to the default values.

**Comment by hannakoppelaar** [comment]
I'm done reviewing the code, but I'm having trouble running the new binaries. Apparently, this webapp now includes servlet-api-2.5-20081211.jar in the lib directory, which contains an old servlet api, and (on my machine at least) this gets loaded before the newer one that we actually rely on. As a consequence webapp doesn't run. If I build master, everything's fine: servlet-api-2.5-20081211.jar doesn't get added.

**Comment by leefrank9527** [comment]
@hannakoppelaar I double checked the webcurator-webapp-3.0.0-SNAPSHOT.war and the gradle dependencies, and didn't find the servlet-api-2.5-20081211.jar. Would you mind print the gradle dependencies of your environment with command: **./gradlew dependencies > webapp.dep** ?

**Comment by hannakoppelaar** [comment]
@leefrank9527 Sure, my output of `./gradlew dependencies` shows the 2.5 servlet-api as a dependency of core:

...
+--- org.webcurator:webcurator-core:3.0.0-SNAPSHOT
|    +--- org.archive:heritrix:1.14.2-webcuratortool-2.0.2
|    |    +--- net.htmlparser.jericho:jericho-html:2.6.1
|    |    +--- com.sleepycat:je:3.3.62 -> 4.1.6
|    |    +--- commons-httpclient:commons-httpclient:3.1.1-heritrix-1.14.2-webcuratortool-2.0.1 (*)
|    |    +--- commons-io:commons-io:1.3.1 -> 2.4
|    |    +--- commons-lang:commons-lang:2.3 -> 2.6
|    |    +--- commons-logging:commons-logging:1.0.4 -> 1.2
|    |    +--- commons-net:commons-net:1.4.1 -> 2.0
|    |    +--- commons-codec:commons-codec:1.3 -> 1.11
|    |    +--- dnsjava:dnsjava:2.0.6
|    |    +--- org.mortbay.jetty:jetty:4.2.12 -> 6.1.26
|    |    |    +--- org.mortbay.jetty:jetty-util:6.1.26
|    |    |    \--- org.mortbay.jetty:servlet-api:2.5-20081211
...

Though, strangely, if I run the command in the webcurator-core dir the module does not show up.

**Comment by hannakoppelaar** [comment]
Interestinly, if I comment out this line in build.gradle (in webapp)

`implementation 'org.archive.heritrix:heritrix-engine:3.4.0-SNAPSHOT'`

the jetty servlet-api dependency disappears.

**Comment by leefrank9527** [comment]
@hannakoppelaar 'org.archive.heritrix:heritrix-engine:3.4.0-SNAPSHOT' is used to validate the profile, I added it to fix an issue. I couldn't find the lib in webcurator-core either.   Is it possible to re-build and re-install webcurator-core before building webcurator-webapp?

**Comment by hannakoppelaar** [comment]
@leefrank9527 Apparently, I had an old heritrix 3.4.0-SNAPSHOT in my m2 repo that triggered the inclusion of the old servlet api. After deleting it, the servlet 2.5 api is gone from the dependency tree.

This does bring us to another issue though: how is the heritrix 3.4.0 artefact supposed to be included by gradle? It's not in any of the configured repos, is it? Should it be added to the install_maven_dependencies script? Or is there a repo somewhere with Heritrix builds? Obviously, I know how to build Heritrix, but our build process should not rely on that knowledge ;)

**Comment by leefrank9527** [comment]
@hannakoppelaar You are right, this is a potential issue. Gradle will cascadely manage dependencies. Heritrix 3.4.0-SNAPSHOT is dependent on by heritrix-engine, openwayback and other aitifacts. We can't abandon this with"exclude" announcement in the build.gradle. Another trouble is gradle will try to find the dependencies from local repository (.m2). So if there is an old jars in .m2, it's hard for us to ignore it automaticly.

To make a completely clean build. I think we could remove all aitificts in .m2, and inital them with webcurator-legacy-lib-dependencies/. But if we have other projects shared the .m2, that would be inconvenient.

**Comment by obrienben** [comment]
@hannakoppelaar H3 looks to now be in maven central - https://mvnrepository.com/artifact/org.archive.heritrix/heritrix-engine. So it should be ok to get pulled in via Gradle right, instead of webcurator-legacy-lib-dependencies?

**Comment by hannakoppelaar** [comment]
@obrienben yeah, I've now built it from scratch (starting with an empty m2 repo) using heritrix-engine:3.4.0-20190205 from maven central. I.e. I've changed the heritrix dependency in webapp's build.gradle to

`implementation 'org.archive.heritrix:heritrix-engine:3.4.0-20190205'`

And it builds okay. The sad news is: I'm still getting the old jetty servlet api in my build. I'm using the same gradle version as @leefrank9527: 5.6 with JDK 1.8.0_275.

**Pull request merged**